src/main/scala/de/retest/guistatemachine/api/impl/serialization/GuiStateMachineJavaObjectStreamSerializer.scala

```diff
@@ -0,0 +1,35 @@
+package de.retest.guistatemachine.api.impl.serialization
+
+import java.io.{FileInputStream, FileOutputStream, ObjectInputStream, ObjectOutputStream}
+
+import de.retest.guistatemachine.api.impl.GuiStateMachineImpl
+import de.retest.guistatemachine.api.{GuiStateMachine, GuiStateMachineSerializer}
+
+case class GuiStateMachineJavaObjectStreamSerializer(guiStateMachine: GuiStateMachine) extends GuiStateMachineSerializer {
+
+  /**
+    * Stores the state machine on the disk.
+    * Persistence can be useful when the state machines become quite big and the generation/modification is interrupted
+    * and continued later.
+    *
+    * @param filePath The file which the state machine is stored into.
+    */
+  override def save(filePath: String): Unit = {
+    val oos = new ObjectOutputStream(new FileOutputStream(filePath))
+    oos.writeObject(guiStateMachine)
+    oos.close()
```

Comment:
If an exception occurs during `writeObject(Object)`, `close()` is not invoked and the resource is never released, so it should be wrapped inside a `finally` block. (Same below.)